src/rmarkdown/manager.ts

```diff
@@ -0,0 +1,146 @@
+import * as util from '../util';
+import * as vscode from 'vscode';
+import * as cp from 'child_process';
+import path = require('path');
+
+interface IKnitArgs {
+	filePath: string;
+	fileName: string;
+	cmd: string;
+	rCmd?: string;
+	rOutputFormat?: string;
+	callback: (...args: unknown[]) => boolean;
+	onRejection?: (...args: unknown[]) => unknown;
+}
+
+export interface IKnitRejection {
+	cp: cp.ChildProcessWithoutNullStreams;
+	wasCancelled: boolean;
+}
+
+const rMarkdownOutput: vscode.OutputChannel = vscode.window.createOutputChannel('R Markdown');
+
+export abstract class RMarkdownManager {
+	protected rPath: string;
+	protected rMarkdownOutput: vscode.OutputChannel = rMarkdownOutput;
+	// uri that are in the process of knitting
+	// so that we can't spam the knit/preview button
+	protected busyUriStore: Set<string> = new Set<string>();
+
+	public async init(): Promise<void> {
+		this.rPath = await util.getRpath(true);
```

Comment:
If `rPath` is determined in `init()`, then chaning R paths in settings will not have effect before restart. Maybe we should call `util.getRpath()` each time we use it?